### PR TITLE
Add PHP_CodeSniffer to Composer along with some initial rules

### PR DESCRIPTION
**Pull request by Viper007Bond**
This adds PHP_CodeSniffer to Composer along with some initial rules:

* [Rules for core WordPress](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards)
* @sirbrillig's [VariableAnalysis rules](https://github.com/sirbrillig/phpcs-variable-analysis)
* [Rules for ensuring PHP 5.2+ compatibility](https://github.com/wimg/PHPCompatibility)

**These rules are not actually being enforced anywhere**, such as in CI tests, rather this is an attempt to standardize coding rules for the project and to allow for easily enabling of linting in IDEs such as PHPStorm.

TL;DR: This should help us write better code.

### CODE REVIEW

**Comment by Viper007Bond** [comment]
A sample example:

phpstorm64_2018-01-29_23-16-08

File: .svnignore [hunk 1]
```diff
@@ -31,4 +33,4 @@ _inc/build/*.js.map
 _inc/build/*.css.map
 _inc/build/maps
 .codeclimate.yml
-yarn.lock
+yarn.lock
```
Comment (ebinnion) [inline .svnignore]: nit: There should probably be an empty newline at the end of this file

**Comment by ebinnion** [comment]
This is the error I get before this PR:

screen shot 2018-01-30 at 4 15 58 pm

This is after checking out this PR:

screen shot 2018-01-30 at 4 16 27 pm

Works well.
